service/resource/vaultcrt/update.go

```diff
@@ -25,10 +56,80 @@ func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desire
 	return patch, nil
 }
 
-func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
-	return nil
+func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	currentSecret, err := toSecret(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredSecret, err := toSecret(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out if the secret has to be updated")
+
+	var secretToUpdate *apiv1.Secret
+	{
+		TTL, err := time.ParseDuration(key.CrtTTL(customObject))
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+
+		renew, err := r.shouldCertBeRenewed(currentSecret, TTL, r.expirationThreshold)
+		if IsMissingAnnotation(err) {
```

Comment:
If the annotation is missing, could we add it?

Reply:
We add it with new clusters. I think there will be a transition period like in the kvm-operator with the version bundle versions tracked in the deployments.

Reply:
My thinking was we could avoid the transition period. No strong opinion.